### PR TITLE
Allow FROM to be overriden with from option

### PR DESCRIPTION
**Pull request by TomSweeneyRedHat**
These changes will allow the "FROM" statement in a Containerfile
to be overridden with a new `--from` option.  If I have this Dockerfile.fedora

```
FROM fedora
```

This command will instead build an alpine image:

```
# buildah bud --from=alpine -t tom -f ~/Dockerfile.fedora .
STEP 1: FROM alpine
Completed short name "alpine" with unqualified-search registries (origin: /etc/containers/registries.conf)
Getting image source signatures
Copying blob 188c0c94c7c5 done
Copying config d6e46aa247 done
Writing manifest to image destination
Storing signatures
STEP 2: COMMIT tom
--> d6e46aa2470
d6e46aa2470df1d32034c6707c8041158b652f38d2a9ae3d7ad7e7532d22ebe0
```

Addresses: #2404

Signed-off-by: TomSweeneyRedHat <tsweeney@redhat.com>

#### What type of PR is this?

> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test 
> /kind feature
> /kind flake
> /kind other

#### Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: imagebuildah/executor.go
Comment (rhatdan) [inline imagebuildah/executor.go:50]: Where are these documented?  Doesn't this new one need documentation in a man page.  Currently this seems like hidden magic?

Comment (rhatdan) [reply imagebuildah/executor.go:50]: https://docs.docker.com/engine/reference/commandline/build/

Could you craft a section in the buildah man page from the above to document these as well as your new FROM arg.

**Comment by rhatdan** [comment]
whitespace

File: docs/buildah-bud.md
Comment (rhatdan) [inline docs/buildah-bud.md:676]: What about FROM?

**Review by rhatdan — APPROVED** [review]
Add FROM documentation to man page then LGTM

File: docs/buildah-bud.md
Comment (TomSweeneyRedHat) [reply docs/buildah-bud.md:676]: next paragraph, slightly different handling as these are all specified like `ENV HTTP_PROXY="http://127.0.0.1:8321", but let me tease that out a bit more to be clear.

**Comment by TomSweeneyRedHat** [comment]
Tried to make the man page a bit more clear on the Build Time Variables and finally chased down the white space issues.

**Comment by rhatdan** [comment]
LGTM
@nalind PTAL

**Comment by nalind** [comment]
This seems to conflict with the way imagebuilder does it:  as a dedicated CLI option which overrides the base image for only the first stage.  This lets it avoid adding a new predefined ARG.  An API caller isn't going to know that we're doing adding an ARG, and I'm worried that this could be hard to troubleshoot later on.

**Comment by rhatdan** [comment]
@nalind Do you have an alternative idea?

**Comment by nalind** [comment]
Instead of tunneling it through an ARG, make it a field in BuildOptions that's controlled by a CLI flag, and when we alter the parsed tree in `imagebuildah.Executor.Build()`, only do so if it's the first stage.

File: docs/buildah-bud.md
Comment (rhatdan) [inline docs/buildah-bud.md:678]: Does this make sense as the default.  I would have thought the more likely request would have been to only change the first?  What happens when they FROMs inside of the Containerfile refer back to previous steps?

Comment (TomSweeneyRedHat) [reply docs/buildah-bud.md:678]: I'll make it change just the first instance then rather than all.  Any thought on the CLI flag?  `-j-first-from` perhaps?

Comment (rhatdan) [reply docs/buildah-bud.md:678]: I would just change the first FROM for  now, and ignore the rest, no new flag.  I think that was what was asked for.  Wait until someone else asks for changing all of the FROMs.  At that point you probably need to rewrite your containerfile.

Comment (nalind) [reply docs/buildah-bud.md:678]: The option for imagebuilder is `-from`, and it's specified using `from:` [in OpenShift](https://docs.openshift.com/container-platform/4.6/builds/build-strategies.html#builds-strategy-docker-from-image_build-strategies), so `--from` seems fine.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat Needs rebase.

File: docs/buildah-bud.md
Comment (TomSweeneyRedHat) [reply docs/buildah-bud.md]: I've changed it to a `--from`  option per @nalind's last comment.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan @nalind reworked and repushed.  Ready for the next round of reviews.  I've changed from using a Build Time variable passed in with a build-arg to creating a new `--from` option.

**Comment by rhatdan** [comment]
LGTM if you can get tests to pass.

File: imagebuildah/executor.go
Comment (nalind) [inline imagebuildah/executor.go:252]: Why set it here instead of on line 233?

Comment (TomSweeneyRedHat) [reply imagebuildah/executor.go:252]: Because at the time my noggin was thinking I needed to set it to "" by default, but that I already took care of in the declaration in pkg/cli/common.go.  Good thought, I'll change it.

**Comment by rhatdan** [comment]
--from needs to be added to completions.

**Comment by TomSweeneyRedHat** [comment]
@rhatdan, `--from` added to the completions and now this has all happy green test buttons.

**Comment by rhatdan** [comment]
@containers/podman-maintainers PTAL
Lets get this merged.
@saschagrunert @giuseppe @vrothberg @nalind @ashley-cui @umohnani8 PTAL

**Review by ashley-cui — COMMENTED** [review]
LGTM

**Comment by kitty-catt** [comment]
I just ran the --from flag on buildah provided by registry.redhat.io/rhel8/buildah. The --from option is not there yet. 

Is there any idea when this option becomes available in registry.redhat.io/rhel8/buildah? 

Do I have an alternative?

I have attached the output of the buildah version and help commands:
[buildah-output.txt](https://github.com/containers/buildah/files/6378077/buildah-output.txt)

**Comment by nalind** [comment]
The feature landed in the v1.19 release.  The `quay.io/buildah/stable` image, which is built similarly, is currently at v1.19.8.  I expect an update to `registry.redhat.io/rhel8/buildah` for RHEL 8.4 will update it to something close to that version, depending on the timing of it.

**Comment by kitty-catt** [comment]
Thank you nalind. It works when I use quay.io/buildah/stable (in a tekton task on openshift)! I intend to switch back to redhat once they have put it in.

**Comment by TomSweeneyRedHat** [comment]
@fatherlinux pinging you in case there's anything that needs to happen to update the Buildah image on the RHEL registries.

**Pull request merged**
